### PR TITLE
Add client plugin support

### PR DESCRIPTION
**Pull request by McInkay**

### CODE REVIEW

**Comment by McInkay** [comment]
Just to be clear, this is purely client plugins in general, with no API. It's probably simplest to agree on how we do client plugins separately from anything else.

This does have a use to merge in on it's own as well because people can create plugins that touch the dom and do some messing around that way, which will give people somewhere to start, and the APIs for our Javascript can come later.

File: client/index.html.tpl [hunk 1]
```diff
@@ -7,6 +7,11 @@
 
 	<link rel="preload" as="script" href="js/bundle.vendor.js">
 	<link rel="preload" as="script" href="js/bundle.js">
+	<!--we import the package manager first, so that it's cached for plugins and plugins can't override the object -->
```
Comment (xPaw) [inline client/index.html.tpl]: Why are they included in head and not after lounges code?

File: client/js/packages/packageManager.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const contextMenu = window.TheLounge.contextMenu;
```
Comment (McInkay) [inline client/js/packages/packageManager.js]: This shouldn't have been left in btw.

File: client/index.html.tpl (hunk 1 above)
Comment (McInkay) [reply client/index.html.tpl]: It is after lounges code, it's after the bundle.

Comment (xPaw) [reply client/index.html.tpl]: This is preload. This should be at the bottom of the file

Comment (McInkay) [reply client/index.html.tpl]: Oh I missed that. Alrighty. I need to re-test this and make sure things actually work anyway, but yeah good point.

File: client/js/packages/packageApi.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Our public API
+window.TheLounge = {
+};
```
Comment (McInkay) [inline client/js/packages/packageApi.js]: I think I should probably add at least 1 tiny thing in here so I can test this stuff works, something like ... maybe ... the version of the lounge that we are running against maybe?

Comment (lol768) [reply client/js/packages/packageApi.js]: >maybe ... the version of the lounge that we are running against maybe

Would be useful, IMO

**Comment by McInkay** [comment]
I think I'll close this just now, but keep the code. I think I should probably focus on building up the server-side api before getting this in. We'll get there.

**Comment by astorije** [comment]
@McInkay, I'll delete the branch to remove clutter, but that doesn't mean the code goes away: closed PRs are like "archived" so code stays forever, and you can re-open the branch whenever you're ready to pick this up. I'm going to do the exact same thing in #1950 :)

**Pull request closed without merging**